engine/src/main/java/org/terasology/config/SystemConfig.java

```diff
@@ -16,99 +16,63 @@
-    private int maxUnloadedChunksPercentageTillSave;
-    private boolean debugEnabled;
-    private boolean monitoringEnabled;
-    private boolean writeSaveGamesEnabled;
-    private long chunkGenerationFailTimeoutInMs;
-    private String locale;
-
-    public long getDayNightLengthInMs() {
-        return dayNightLengthInMs;
-    }
-
-    public void setDayNightLengthInMs(long dayNightLengthInMs) {
-        this.dayNightLengthInMs = dayNightLengthInMs;
-    }
-
-    public int getMaxThreads() {
-        return maxThreads;
-    }
-
-    public void setMaxThreads(int maxThreads) {
-        this.maxThreads = maxThreads;
-    }
-
-    public int getMaxSecondsBetweenSaves() {
-        return maxSecondsBetweenSaves;
-    }
-
-    public void setMaxSecondsBetweenSaves(int maxSecondsBetweenSaves) {
-        this.maxSecondsBetweenSaves = maxSecondsBetweenSaves;
-    }
-
-    public int getMaxUnloadedChunksPercentageTillSave() {
-        return maxUnloadedChunksPercentageTillSave;
-    }
-
-    public void setMaxUnloadedChunksPercentageTillSave(int maxUnloadedChunksPercentageTillSave) {
-        this.maxUnloadedChunksPercentageTillSave = maxUnloadedChunksPercentageTillSave;
-    }
-
-    public boolean isDebugEnabled() {
-        return debugEnabled;
-    }
-
-    public void setDebugEnabled(boolean debugEnabled) {
-        this.debugEnabled = debugEnabled;
-    }
-
-    public boolean isMonitoringEnabled() {
-        return monitoringEnabled;
-    }
-
-    public void setMonitoringEnabled(boolean monitoringEnabled) {
-        this.monitoringEnabled = monitoringEnabled;
-    }
-
-    public boolean isWriteSaveGamesEnabled() {
-        String property = System.getProperty(SAVED_GAMES_ENABLED_PROPERTY);
-        if (property != null) {
-            return Boolean.parseBoolean(property);
-        }
-        return writeSaveGamesEnabled;
-    }
-
-    public void setWriteSaveGamesEnabled(boolean writeSaveGamesEnabled) {
-        this.writeSaveGamesEnabled = writeSaveGamesEnabled;
-    }
-
-    public long getChunkGenerationFailTimeoutInMs() {
-        return chunkGenerationFailTimeoutInMs;
-    }
-
-    public void setChunkGenerationFailTimeoutInMs(long chunkGenerationFailTimeoutInMs) {
-        this.chunkGenerationFailTimeoutInMs = chunkGenerationFailTimeoutInMs;
-    }
-
-    public Locale getLocale() {
-        if (locale == null) {
-            setLocale(Locale.getDefault(Category.DISPLAY));
-        }
-        return Locale.forLanguageTag(locale);
-    }
-
-    public void setLocale(Locale locale) {
-        this.locale = locale.toLanguageTag();
-    }
+    public final Setting<Long> dayNightLengthInMs = setting(
+            type(Long.class),
+            defaultValue(1800000L),
+            constraint(new NumberRangeConstraint<>(0L, Long.MAX_VALUE, false, false))
+    );
+    public final Setting<Integer> maxThreads = setting(
+            type(Integer.class),
+            defaultValue(Runtime.getRuntime().availableProcessors() - 1),
```

Comment:
Where do these default values come from?

Reply:
This value from recommendations for threading(Google) and i saw it in issue about FlexableConfig.
(Default value in config is 2)

All other value from config.

Reply:
On single-core systems, this comes out to zero. :boom:

Reply:
Ok. Will fixed.... Where are you found single core? VM?